hoss/dimension_utilities.py

```diff
@@ -55,54 +65,329 @@ def is_index_subset(message: Message) -> bool:
     )
 
 
-def prefetch_dimension_variables(
+def get_prefetch_variables(
     opendap_url: str,
     varinfo: VarInfoFromDmr,
     required_variables: Set[str],
     output_dir: str,
     logger: Logger,
     access_token: str,
     config: Config,
-) -> str:
-    """Determine the dimensions that need to be "pre-fetched" from OPeNDAP in
+) -> tuple[str, set[str]] | str:
+    """Determine the variables that need to be "pre-fetched" from OPeNDAP in
     order to derive index ranges upon them. Initially, this was just
     spatial and temporal dimensions, but to support generic dimension
     subsets, all required dimensions must be prefetched, along with any
     associated bounds variables referred to via the "bounds" metadata
-    attribute.
+    attribute. In cases where dimension variables do not exist, coordinate
+    variables will be prefetched and used to calculate dimension-scale values
 
     """
-    required_dimensions = varinfo.get_required_dimensions(required_variables)
-
-    # Iterate through all requested dimensions and extract a list of bounds
-    # references for each that has any. This will produce a list of lists,
-    # which should be flattened into a single list and then combined into a set
-    # to remove duplicates.
-    bounds = set(
-        flatten_list(
-            [
-                list(varinfo.get_variable(dimension).references.get('bounds'))
-                for dimension in required_dimensions
-                if varinfo.get_variable(dimension).references.get('bounds') is not None
-            ]
-        )
-    )
-
-    required_dimensions.update(bounds)
+    required_variables = varinfo.get_required_dimensions(required_variables)
+    if required_variables:
+        bounds = varinfo.get_references_for_attribute(required_variables, 'bounds')
+        required_variables.update(bounds)
+    else:
+        coordinate_variables = get_coordinate_variables(varinfo, required_variables)
+        if coordinate_variables:
+            required_variables = set(coordinate_variables)
 
     logger.info(
         'Variables being retrieved in prefetch request: '
-        f'{format_variable_set_string(required_dimensions)}'
+        f'{format_variable_set_string(required_variables)}'
     )
 
-    required_dimensions_nc4 = get_opendap_nc4(
-        opendap_url, required_dimensions, output_dir, logger, access_token, config
+    required_variables_nc4 = get_opendap_nc4(
+        opendap_url, required_variables, output_dir, logger, access_token, config
     )
 
     # Create bounds variables if necessary.
-    add_bounds_variables(required_dimensions_nc4, required_dimensions, varinfo, logger)
+    add_bounds_variables(required_variables_nc4, required_variables, varinfo, logger)
+    return required_variables_nc4
+
+
+def get_override_projected_dimension_name(
```

Comment:
Maybe this would be a bit easier to digest if all these new functions that are specific to this edge-case are in their own file?

Reply:
I could do that

Reply:
created a coordinates_utilities.py - which has most of the new methods

Reply:
_Awesome_, thanks for this!